tensorflow/c/eager/c_api_unified_experimental_test.cc

```diff
@@ -169,6 +331,112 @@ TEST_P(UnifiedCAPI, TestBasicGraph) {
   TF_DeleteExecutionContext(eager_execution_ctx);
 }
 
+// Graph Tracing for MatMul
+TEST_P(UnifiedCAPI, TestBasicGraphMatMul) {
+  std::unique_ptr<TF_Status, decltype(&TF_DeleteStatus)> status(
+      TF_NewStatus(), TF_DeleteStatus);
+  
+  // Start a new function / execution context.
+  string fn_name = "matrix_multiply";
+  TF_ExecutionContext* graph_ctx =
+      TF_CreateFunction(fn_name.c_str(), status.get());
+  ASSERT_EQ(TF_OK, TF_GetCode(status.get())) << TF_Message(status.get());
+
+  auto* placeholder_t =
+      TF_AddFunctionParameter(graph_ctx, TF_FLOAT, status.get());
+  ASSERT_EQ(TF_OK, TF_GetCode(status.get())) << TF_Message(status.get());
+
+  // Build an abstract operation.
+  auto* matmul_op = TF_NewAbstractOp(graph_ctx);
+  TF_AbstractOpSetOpType(matmul_op, "MatMul", status.get());
+  ASSERT_EQ(TF_OK, TF_GetCode(status.get())) << TF_Message(status.get());
+  TF_AbstractOpSetOpName(matmul_op, "my_matmul", status.get());
+  ASSERT_EQ(TF_OK, TF_GetCode(status.get())) << TF_Message(status.get());
+
+  // Build inputs and outputs.
+  TF_AbstractTensor* inputs[2] = {placeholder_t, placeholder_t};
+  TF_OutputList* mm_outputs = TF_NewOutputList();
+
+  // Execute.
+  TF_ExecuteOperation(matmul_op, 2, inputs, mm_outputs, graph_ctx, status.get());
+  ASSERT_EQ(TF_OK, TF_GetCode(status.get())) << TF_Message(status.get());
+
+  // Clean up operation and inputs.
+  TF_DeleteAbstractOp(matmul_op);
+
+  TF_AbstractFunction* func =
+      TF_FinalizeFunction(graph_ctx, mm_outputs, status.get());
```

Comment:
TF_OutputList does not own the underlying abstract tensors. You would need to delete the underlying tensors in `mm_outputs`. 

`TF_DeleteAbstractTensor(TF_OutputListGet(mm_outputs, 0));`

Reply:
Actually, the result tensor is stored in **final_result** which is freed after **mm_outputs**. Just confirmed this was the case by trying to call both **TF_DeleteAbstractTensor(TF_OutputListGet(mm_outputs, 0));** and **TF_DeleteAbstractTensor(final_result);**, which results in an error. I'll move the **TF_DeleteAbstractTensor(final_result);** a few lines up so it's clearer.

Reply:
We reuse `mm_outputs` for storing both the function outputs (here) and for storing the outputs of executing the function below. `TF_OutputListSetNumOutputs(mm_outputs, 1, status.get())` resets the underlying vector of graph tensors but does not actually delete them so those are the ones that would would leak. 

We should probably think about how to better manage the ownerships of pointers here. The current setup, although very explicit, may be prone to errors..